test/common/upstream/upstream_impl_test.cc

```diff
@@ -1625,14 +1685,27 @@ TEST_F(ClusterInfoImplTest, ExtensionProtocolOptionsForFilterWithOptions) {
       envoy.test.filter: { option: "value" }
   )EOF";
 
-  auto cluster = makeCluster(yaml);
+  std::vector<std::unique_ptr<StrictDnsClusterImpl>> clusters;
```

Comment:
Can you add some comment explaining what you're testing here? It's unclear without reading deeply whether we're covering all the permutations of {no filter, network filter, http filter} for a given protocol options slot.